File: lib/domain/cultivation_plan/gateways/plan_allocation_adjust_debug_dump_null_gateway.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Gateways
      class PlanAllocationAdjustDebugDumpNullGateway < PlanAllocationAdjustDebugDumpGateway
        def dump_payload!(current_allocation:, moves:, fields:, crops:)
          nil
        end
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/gateways/plan_allocation_adjust_debug_dump_null_gateway.rs
//! Ruby: `Domain::CultivationPlan::Gateways::PlanAllocationAdjustDebugDumpNullGateway`

use super::PlanAllocationAdjustDebugDumpGateway;
use serde_json::Value;

/// No-op debug dump (production).
pub struct PlanAllocationAdjustDebugDumpNullGateway;

impl PlanAllocationAdjustDebugDumpGateway for PlanAllocationAdjustDebugDumpNullGateway {
    fn dump_payload(
        &self,
        _current_allocation: &Value,
        _moves: &[Value],
        _fields: &[Value],
        _crops: &[Value],
    ) {
    }
}
